### PR TITLE
testcase:boot_menu_list is NULL 

### PR DESCRIPTION
**Pull request by xiangchunfu**
1.As sendkey "esc" is later,So boot_menu_list is NULL
2.Get boot_menu_list via sendkey "esc" during booting.
id:1509197
Signed-off-by: Xiangchun Fu <xfu@redhat.com>

### CODE REVIEW

**Comment by xiangchunfu** [comment]
(1/1) smp_4.2048m.repeat1.run_test.Host_RHEL.m7.u4.qcow2.virtio_blk.up.virtio_net.RHEL.7.4.x86_64.io-github-autotest-qemu.seabios: PASS (400.88 s)

**Comment by quanwenli** [comment]
Acked-by: Wenli Quan wquan@redhat.com

JOB ID     : d2ff83b1396d9df9e2083d4b640366ddee8d7edf
JOB LOG    : /root/avocado/job-results/job-2017-11-06T02.12-d2ff83b/job.log
 (1/1) smp_8.8192m.repeat1.run_test.Host_RHEL.m7.u4.qcow2.virtio_blk.up.virtio_net.RHEL.7.4.x86_64.io-github-autotest-qemu.seabios: PASS (137.17 s)

**Comment by yanan-fu** [comment]
@xiangchunfu  do we have any scenario that send boot_menu_key "ESC" without restart VM ?

**Comment by yanan-fu** [comment]
@xiangchunfu  If not,  move  send boot_menu_key into `if restart_key:`  is ok.  But, the whole process  need to be adjusted.  IMO,  the following code can not be put after "ESC"  be sent.   
'''
     if not (boot_menu_hint and utils_misc.wait_for(boot_menu, timeout, 1)):
         raise error.TestFail("Could not get boot menu message.")
'''

**Comment by xiangchunfu** [comment]
@yanan-fu  Thanks Yanan.  
1. There is not this scenario that send boot_menu_key "ESC" without restart VM.
2. According to your suggestion.  Don't change the original logic.  So have updated patch. please review it again.

 (1/1) smp_4.512m.repeat1.run_test.Host_RHEL.m7.u5.qcow2.virtio_blk.up.virtio_net.RHEL.7.4.x86_64.io-github-autotest-qemu.seabios: PASS (412.11 s)

**Comment by yanan-fu** [comment]
@xiangchunfu  L72~L73 can be removed now,

File: qemu/tests/seabios.py [hunk 1]
```diff
@@ -62,18 +63,24 @@ def info_check():
...
         seabios_text = get_output(seabios_session)
         headline = seabios_text.split("\n")[0] + "\n"
         headline_count = seabios_text.count(headline)
 
         vm.send_key(restart_key)
 
+        if not (boot_menu_hint and utils_misc.wait_for(boot_menu, timeout, 1)):
+            raise error.TestFail("Could not get boot menu message.")
+        # Send boot menu key in monitor.
+        time.sleep(1)
```
Comment (yanan-fu) [inline qemu/tests/seabios.py]: The `sleep(1) `here can be removed too, i think.

File: qemu/tests/seabios.py [hunk 2]
```diff
@@ -62,18 +63,24 @@ def info_check():
...
-            raise error.TestFail("Could not get boot menu message.")
-
         seabios_text = get_output(seabios_session)
         headline = seabios_text.split("\n")[0] + "\n"
         headline_count = seabios_text.count(headline)
 
         vm.send_key(restart_key)
 
+        if not (boot_menu_hint and utils_misc.wait_for(boot_menu, timeout, 1)):
+            raise error.TestFail("Could not get boot menu message.")
```
Comment (xiagao) [inline qemu/tests/seabios.py]: use test.fail()

File: qemu/tests/seabios.py [hunk 3]
```diff
@@ -62,18 +63,24 @@ def info_check():
             err_msg += " is %s" % get_output(vm.serial_console)
             raise error.TestFail(err_msg)
 
+    if not (boot_menu_hint and utils_misc.wait_for(boot_menu, timeout, 1)):
+        raise error.TestFail("Could not get boot menu message.")
```
Comment (xiagao) [inline qemu/tests/seabios.py]: pls use test.fail() to replace raise error.TestFail

File: qemu/tests/seabios.py [hunk 4]
```diff
@@ -84,12 +91,6 @@ def reboot_check():
 
     error.context("Display and check the boot menu order", logging.info)
```
Comment (sitoliu) [inline qemu/tests/seabios.py]: This line can be a prompt for `vm.send_key(boot_menu_key)`, you could put it before line82.

Comment (xiangchunfu) [reply qemu/tests/seabios.py]: @sitoliu  boot menu order is displayed after vm.send_key.  so if put it before line82.  it doesn't works.

File: qemu/tests/seabios.py (hunk 2 above)
Comment (xiangchunfu) [reply qemu/tests/seabios.py]: updated

File: qemu/tests/seabios.py (hunk 3 above)
Comment (xiangchunfu) [reply qemu/tests/seabios.py]: @xiagao  updated

File: qemu/tests/seabios.py (hunk 1 above)
Comment (xiangchunfu) [reply qemu/tests/seabios.py]: removed. It still works.

File: qemu/tests/seabios.py [hunk 5]
```diff
@@ -60,55 +60,54 @@ def info_check():
         if not utils_misc.wait_for(info_check, timeout, 1):
             err_msg = "Cound not get sgabios message. The output"
             err_msg += " is %s" % get_output(vm.serial_console)
-            raise error.TestFail(err_msg)
+            raise test.fail(err_msg)
```
Comment (sitoliu) [inline qemu/tests/seabios.py]: `raise test.fail()` -> "test.fail()"

File: qemu/tests/seabios.py [hunk 6]
```diff
@@ -60,55 +60,54 @@ def info_check():
...
+        error_context.context("Restart vm and check it's ok", logging.info)
 
         seabios_text = get_output(seabios_session)
         headline = seabios_text.split("\n")[0] + "\n"
         headline_count = seabios_text.count(headline)
 
         vm.send_key(restart_key)
 
+        if not (boot_menu_hint and utils_misc.wait_for(boot_menu, timeout, 1)):
+            raise test.fail("Could not get boot menu message.")
```
Comment (sitoliu) [inline qemu/tests/seabios.py]: Should we make this prompt different from line 66, like `Could not get boot menu message after rebooting.` ?

Comment (xiangchunfu) [reply qemu/tests/seabios.py]: @sitoliu  good idea,  updated

File: qemu/tests/seabios.py (hunk 5 above)
Comment (xiangchunfu) [reply qemu/tests/seabios.py]: updated

**Comment by sitoliu** [comment]
smp_16.8192m.repeat1.run_test.Host_RHEL.m7.u4.qcow2.virtio_scsi.up.virtio_net.RHEL.7.4.x86_64.myprovider.seabios: PASS (419.17 s)

Acked-by: Sitong Liu siliu@redhat.com

**Comment by luckyh** [comment]
Please rewrite the commit with the following format:

*testcase: short description*

*long description*

**Comment by xiangchunfu** [comment]
@luckyh Just saw your comment.  I modified description.  please check it again.

**Comment by xiangchunfu** [comment]
@luckyh  done

**Comment by luckyh** [comment]
Thanks all.

**Pull request merged**